`queryFn` field became required in `UseInfiniteQueryOptions`

### Describe the bug

Hi,

It seems that in a recent release, the `queryFn` field of type `UseInfiniteQueryOptions` (from "@tanstack/react-query") became required (was optional in previous releases).

https://github.com/TanStack/query/commit/a991d922e84bcf9ac46b7e23f0f72902166e55cf#diff-ac3343c834b1a6ecb51aa339ab1cf591e1f595d9edfed5ee0d0bc85fd172e66fR59

However, according to the docs, the field is only required if no default query function is defined. https://tanstack.com/query/v5/docs/framework/react/reference/useInfiniteQuery

Is this change intentional? Thanks!

### Your minimal, reproducible example

https://stackblitz.com/edit/vitejs-vite-ufudqn

### Steps to reproduce

1. Open a terminal tab
2. Run "npm run build"
3. Notice the type checking error in "src/App.tsx"

```
src/App.tsx:10:28 - error TS2345: Argument of type '{ queryKey: string[]; initialPageParam: () => {}; getNextPageParam: () => {}; }' is not assignable to parameter of type 'UseSuspenseInfiniteQueryOptions<unknown, Error, InfiniteData<unknown, unknown>, unknown, string[], {}>'.
  Property 'queryFn' is missing in type '{ queryKey: string[]; initialPageParam: () => {}; getNextPageParam: () => {}; }' but required in type 'UseSuspenseInfiniteQueryOptions<unknown, Error, InfiniteData<unknown, unknown>, unknown, string[], {}>'.

 10   useSuspenseInfiniteQuery({
                               ~
 11     queryKey: ['myQuery'],
    ~~~~~~~~~~~~~~~~~~~~~~~~~~
... 
 14     //queryFn: () => ({})
    ~~~~~~~~~~~~~~~~~~~~~~~~~
 15   });
```

### Expected behavior

Compiles successfully

### How often does this bug happen?

Every time

### Screenshots or Videos

_No response_

### Platform

All platforms

### Tanstack Query adapter

react-query

### TanStack Query version

5.59.13

### TypeScript version

5.5.4

### Additional context

_No response_

--- a/packages/react-query/src/infiniteQueryOptions.ts
+++ b/packages/react-query/src/infiniteQueryOptions.ts
@@ -48,7 +48,7 @@ export type UnusedSkipTokenInfiniteOptions<
   >,
   'queryFn'
 > & {
-  queryFn: Exclude<
+  queryFn?: Exclude<
     UseInfiniteQueryOptions<
       TQueryFnData,
       TError,

--- a/packages/react-query/src/queryOptions.ts
+++ b/packages/react-query/src/queryOptions.ts
@@ -29,7 +29,7 @@ export type UnusedSkipTokenOptions<
   UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
   'queryFn'
 > & {
-  queryFn: Exclude<
+  queryFn?: Exclude<
     UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>['queryFn'],
     SkipToken
   >

--- a/packages/react-query/src/types.ts
+++ b/packages/react-query/src/types.ts
@@ -50,7 +50,7 @@ export interface UseSuspenseQueryOptions<
     UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
     'queryFn' | 'enabled' | 'throwOnError' | 'placeholderData'
   > {
-  queryFn: Exclude<
+  queryFn?: Exclude<
     UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>['queryFn'],
     SkipToken
   >
@@ -93,7 +93,7 @@ export interface UseSuspenseInfiniteQueryOptions<
     >,
     'queryFn' | 'enabled' | 'throwOnError' | 'placeholderData'
   > {
-  queryFn: Exclude<
+  queryFn?: Exclude<
     UseInfiniteQueryOptions<
       TQueryFnData,
       TError,

